### PR TITLE
PLFM-4044

### PR DESCRIPTION
**Pull request by xschildw**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/wikiV2/V2DBOWikiPageDaoImpl.java [hunk 1]
```diff
@@ -125,16 +128,13 @@
...
-			" on f." + V2_COL_WIKI_MARKDOWN_VERSION + " = m." + V2_COL_WIKI_MARKDOWN_VERSION +
-			" and f." + V2_COL_WIKI_MARKDOWN_ID + " = m." + V2_COL_WIKI_MARKDOWN_ID +
-			" where m." + V2_COL_WIKI_MARKDOWN_ID + " = ? and f." + V2_COL_WIKI_MARKDOWN_VERSION + " is null" ;
+	private static final String SQL_DELETE_WIKI_VERSIONS =
+			"DELETE FROM " + V2_TABLE_WIKI_MARKDOWN +
+				" WHERE " + V2_COL_WIKI_MARKDOWN_ID + " = :wikiPageId AND " +
+				V2_COL_WIKI_MARKDOWN_VERSION + " IN (:versions)";
+	private static final String SQL_UPDATE_WIKI_ETAG = 
+			"UPDATE " + V2_TABLE_WIKI_PAGE + " SET " + V2_COL_WIKI_ETAG + " = ? " +
+			"WHERE " + V2_COL_WIKI_ID + " = ?";
```
Comment (xschildw) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/wikiV2/V2DBOWikiPageDaoImpl.java:138]: Double-check that primary

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java [hunk 2]
```diff
@@ -429,4 +429,49 @@ public WikiPageKey getRootWikiKey(UserInfo user, String ownerId,
...
+		if (wikiId == null) throw new IllegalArgumentException("WikiId cannot be null");
+		if (versionsToDelete == null) throw new IllegalArgumentException("VersionsToDelete cannot be null");
+
+		WikiPageKey key = WikiPageKeyHelper.createWikiPageKey(ownerId, ownerType, wikiId);
+		validateUpdateAccess(user, key);
+
+		String etag = wikiPageDao.lockForUpdate(wikiId);
+		
+		V2WikiPage wiki = wikiPageDao.get(key, null);
+
```
Comment (xschildw) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java:444]: Not needed.

File: integration-test/src/test/java/org/sagebionetworks/IT055WikiPageTest.java [hunk 3]
```diff
@@ -147,6 +152,27 @@ public void testWikiRoundTrip() throws Exception {
...
+		assertNotNull(preHistory);
+		assertNotNull(preHistory.getResults());
+		assertEquals(2, preHistory.getResults().size());
+		
+		String preDeleteEtag = wiki.getEtag();
+		WikiVersionsList versionsToDelete = new WikiVersionsList();
+		List<String> ids = Arrays.asList("0");
+		versionsToDelete.setVersionIds(ids);
+		
+		V2WikiPage pDeletedHistory = synapse.deleteV2WikiVersions(key, versionsToDelete);
```
Comment (john-hill) [inline integration-test/src/test/java/org/sagebionetworks/IT055WikiPageTest.java:166]: return void

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/wikiV2/V2DBOWikiPageDaoImpl.java [hunk 4]
```diff
@@ -778,18 +778,43 @@ private boolean doesExist(String id) {
...
-		ValidateArgument.required(numVersionsToKeep, "numVersionsToKeep");
+	public void deleteWikiVersions(WikiPageKey key, List<String> versionsToDelete) {
+		if (key == null) {
+			throw new IllegalArgumentException("Key cannot be null.");
+		}
+		if (versionsToDelete == null) {
+			throw new IllegalArgumentException("VersionsToDelete cannot be null.");
+		}
+		if (versionsToDelete.size() == 0) {
+			return; // Nothing to do
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/wikiV2/V2DBOWikiPageDaoImpl.java:792]: throw illegal for this case too.

File: lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/wiki/WikiVersionsList.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (john-hill) [inline lib/lib-auto-generated/src/main/resources/schema/org/sagebionetworks/repo/model/wiki/WikiVersionsList.json:1]: Use the existing IdList

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java [hunk 6]
```diff
@@ -429,4 +429,44 @@ public WikiPageKey getRootWikiKey(UserInfo user, String ownerId,
...
+		if (user == null) throw new IllegalArgumentException("User cannot be null");
+		if (key == null) throw new IllegalArgumentException("Key cannot be null");
+		if (versionsToDelete == null) throw new IllegalArgumentException("VersionsToDelete cannot be null");
+
+		validateUpdateAccess(user, key);
+
+		String etag = wikiPageDao.lockForUpdate(key.getWikiPageId());
+		
+		List<V2WikiHistorySnapshot> history = wikiPageDao.getWikiHistory(key, Long.MAX_VALUE, 0L);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/wiki/V2WikiManagerImpl.java:445]: only prohibit delete of the current version (no need to get the history).

**Pull request merged**